src/uri/path.rs

```diff
@@ -170,6 +170,24 @@ impl PathAndQuery {
         ret
     }
 
+    /// Returs the segments of path component
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use http::uri::*;
+    ///
+    /// let path_and_query : PathAndQuery = "/hello/world".parse().unwrap();
+    ///
+    /// assert_eq!(
+    ///     path_and_query.path_segments().collect::<Vec<_>>(),
+    ///     vec!["hello", "world"]
+    /// );
+    /// ```
+    pub fn path_segments(&self) -> str::Split<char> {
```

Comment:
Could you return a new type here? Some type that wraps `str::Split<char>` and implements `Iterator`:

```rust
struct PathSegments(str::Split<char>);
```

This is for forwards compatibility as it allows us to change the implementation details of the iterator.

I'm also on the fence about whether it should be `path_segments` or just `segments`. It is a method on the `PathAndQuery` type, so it could go either way...

Reply:
Sorry to bother you, but I created one. Can you check?